<launch>
  <arg name="launch_main" default="true" />
  <arg name="json" />

  <!-- bin contents data -->
  <node pkg="jsk_2015_05_baxter_apc" type="bin_contents.py" name="bin_contents" respawn="true">
    <param name="json" value="$(arg json)" />
  </node>

  <!-- work order data -->
  <node pkg="jsk_2016_01_baxter_apc" type="work_order.py" name="work_order" respawn="true">
    <rosparam>
        json: $(arg json)
        is_apc2016: false
        gripper: gripper2016
    </rosparam>
  </node>

  <!-- parameter -->
  <param name="/left_hand/state" value="" />
  <param name="/left_hand/target_bin" value="" />
  <param name="/right_hand/state" value="" />
  <param name="/right_hand/target_bin" value="" />

  <node pkg="jsk_rqt_plugins" type="rqt_yn_btn" name="rqt_yn_btn" />

  <!-- object verification -->
  <group ns="left_hand">
    <node name="object_verification"
          pkg="jsk_2015_05_baxter_apc" type="object_verification.py"
          respawn="true">
      <param name="json" value="$(arg json)" />
      <param name="weight" value="$(find jsk_2015_05_baxter_apc)/data/classifier_weight.yml" />
      <remap from="~input/bof" to="bof_object_matcher/output" />
      <remap from="~input/color_hist" to="color_object_matcher/output" />
    </node>
  </group>
  <group ns="right_hand">
    <node name="object_verification"
          pkg="jsk_2015_05_baxter_apc" type="object_verification.py"
          respawn="true">
      <param name="json" value="$(arg json)" />
      <param name="weight" value="$(find jsk_2015_05_baxter_apc)/data/classifier_weight.yml" />
      <remap from="~input/bof" to="bof_object_matcher/output" />
      <remap from="~input/color_hist" to="color_object_matcher/output" />
    </node>
  </group>

  <!-- robot main -->
  <node if="$(arg launch_main)" pkg="jsk_2016_01_baxter_apc" type="main.l" name="left_hand"
    args="&quot;(progn (apc-init :larm-controller) (apc-mainloop :larm))&quot;"
    respawn="true" output="screen">
    <env name="DISPLAY" value="" />
  </node>
  <node if="$(arg launch_main)" pkg="jsk_2016_01_baxter_apc" type="main.l" name="right_hand"
    args="&quot;(progn (apc-init :rarm-controller) (apc-mainloop :rarm))&quot;"
    respawn="true" output="screen">
    <env name="DISPLAY" value="" />
  </node>

  <!-- visualization -->
  <node name="visualize_json"
        pkg="jsk_apc2015_common" type="visualize_json.py"
        args="$(arg json)" />
  <node name="visualize_on_rviz"
        pkg="jsk_2015_05_baxter_apc" type="visualize_on_rviz.py">
    <param name="json" value="$(arg json)" />
  </node>

</launch>
